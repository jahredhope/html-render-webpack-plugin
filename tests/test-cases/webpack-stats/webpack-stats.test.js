const MemoryFS = require("memory-fs");
const webpack = require("webpack");
const path = require("path");

const config = require("./webpack.config");
const getDirContentsSync = require("../../utils/getDirContentsSync");

describe("Render WebpackStats", () => {
  const renderDirectory = path.join(process.cwd(), "dist", "render");

  it("should render a with asset names", async done => {
    const compiler = webpack(config);

    const memoryFs = new MemoryFS();
    compiler.outputFileSystem = memoryFs;

    compiler.run(error => {
      if (error) {
        throw error;
      }
      const contents = getDirContentsSync(renderDirectory, { fs: memoryFs });
      expect(contents).toMatchSnapshot();
      done();
    });
  });
});
