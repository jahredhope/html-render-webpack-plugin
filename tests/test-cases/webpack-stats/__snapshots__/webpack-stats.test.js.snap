// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render WebpackStats should render a with asset names 1`] = `
Object {
  "index.html": "[
  {
    \\"name\\": \\"client\\",
    \\"assets\\": {
      \\"main\\": \\"client-main-ceb3857c2a7596acae6d.js\\"
    }
  },
  {
    \\"name\\": \\"render\\",
    \\"assets\\": {
      \\"main\\": \\"render-main-82d3602c9121065b0ab7.js\\"
    }
  }
]",
}
`;
